### ISSUE TITLE
[Question] How to define same path on multiple collections ?

### ISSUE THREAD

**Issue by qntndev** [gridsome/gridsome#1131]
Hey Gridsome community 👋

Did someone know a workaround to define a path (here `/products/....`) on multiple collections without merging them in Gridsome server ?

```javascript
module.exports = {
  templates: {
    CollectionA: [
      {
        path: '/products/:slug/',
        component: './src/templates/templateA.vue'
      }
    ],
    CollectionB: [
      {
        path: '/products/:slug/',
        component: './src/templates/templateB.vue'
      }
    ]
}
```

If you have already encountered this or have any clue, please let me know 🙏

**Comment by thetre97** [comment]
@qntndev To make sure I understand correctly, you mean CollectionA products never get created?

**Comment by qntndev** [comment]
@thetre97 So the product is created (can query it in GraphQL playground) and the url exists for CollectionA products.

But accessing the path of a CollectionA product is processed by the CollectionB component and the render fails.

<img width="651" alt="Capture d’écran 2020-04-30 à 11 19 18" src="https://user-images.githubusercontent.com/25149311/80694295-98d4bc80-8ad4-11ea-80c7-7574fe55e37d.png">

It set up a demo repo [here](https://github.com/qntndev/gridsome-same-path-collections), so you can clone it and see what happens.

**Comment by thetre97** [comment]
Hey @qntndev , 
Sorry for the delay in replying, did you manage to sort this? 
If not, I would suggest using the [`createPage` API](https://gridsome.org/docs/pages-api/#createpageoptions) to manually create the paths for each product, and you can also specify the template they should use. I can past some demo code if you need any help.

**Comment by quentinlagache** [comment]
Thanks for your input @thetre97 
I managed to get it works with the `createPage` API. The only downside is that I loose the `path` attribute auto populate for collection items in the GraphQL schema. So I need to hardcode the path when needed for a collection item.

**Comment by thetre97** [comment]
@quentinlagache Yes, I have that problem often too. I generally use `onCreateNode` to add a path value to a node, then use that in the `createPage API`, and can also query it in the store.

**Comment by quentinlagache** [comment]
@thetre97 It was my first attempt to solves it. 
But it have to deal with the following error because the path already exist in CollectionB and we try to reference the same path again when we create the page with `createPage`API
```bash
Initializing plugins...
Load sources - 9.77s
Create GraphQL schema - 0.26s
Error: Duplicate key for property path: /products/unique-slug-here
```
I used the following code as suggested.
```javascript
api.createPages(({ getCollection, createPage }) => {
    getCollection('CollectionB')
      .data()
      .forEach(node => {
        createPage({
          path: node.path,
          component: './src/templates/CollectionB.vue',
          context: {
            id: node.id
          }
        });
      });
  });

  api.onCreateNode(options => {
    if (options.internal.typeName === 'CollectionB') {
      options.path = `/products/${options.slug}`;
    }
  });
```
So do you use another key name instead of `path` for this attribute or did I miss something ?

**Comment by quentinlagache** [comment]
Would really love to hear from you on this @thetre97 🤗

**Comment by thetre97** [comment]
@quentinlagache Sorry - that error message seems to suggest that there are two products with the exact same slug? Hence there is then a duplicate key - `/product/123`, `/product/123` etc.

**Comment by thetre97** [comment]
Do you have some sample data I could look at, and could you paste your whole gridsome.server.js (or share a link to a repo)?

**Comment by thetre97** [comment]
Is there any reason you can't just merge the two collections product's in `gridsome.server.js`, as they both have the same path anyway?
